### ISSUE TITLE
cartページへのリンクタグ、aタグだとうまくいかない。カウントの値が０に戻ってしまう

### ISSUE THREAD

**Issue by nobyn0312** [nobyn0312/ec_market_place_user-front#1]
cartへのリンクを、aタグからLinkタグにしたら、jotaiの値が共有できた。

原因がわからないのですが、あまり気にしなくていいでしょうか

**Comment by nobyn0312** [comment]
https://zenn.dev/miyaken0805/articles/c19bada3b1825a
こんな記事ありました

**Comment by yoshidev523** [comment]
そうですね、LinkコンポーネントはSPAで画面遷移するのに対し、aタグはHTMLの標準仕様で `href` 属性に書かれたリンクのHTML情報をサーバーに再度リクエストして取得し画面表示しています。違いはChromeなどのブラウザを見るとわかります。

https://github.com/nobyn0312/ec_market_place_user-front/assets/28664722/5b4e5dc5-95b5-49a2-914e-f25a8c4ae34d

Linkだと更新マークに変化がないのに対し、aタグだと更新マークがバツマークに変わるのが見てわかると思います。バツマークはブラウザがサーバーにリクエストをしてHTML情報を取得している状態を示しています。

LinkはSPA（シングルページアプリケーション）、つまりブラウザは最初の一回しかページ情報をサーバーから取得しておらず、それ以降の画面遷移は全てブラウザ側で行い、あたかも新しいページを取得したかのように見せているだけです。そのためブラウザの更新マークに変化がないにもかかわらずページ遷移ができています。
画面遷移してもCountがリセットされないのは、aタグと違い別のHTML情報を取得しないで画面遷移ができているのでCountのステートが保持されているためです。一方aタグは別のHTML情報を改めて取得しにいってるので今まで保持していたステートが全てリセットされてしまいます。これは、同じページをリロードした時の挙動と同じ原理です。

**Comment by yoshidev523** [comment]
> 原因がわからないのですが、あまり気にしなくていいでしょうか

React.jsなどのSPAライプラりを使う上では気にする必要があります。状態を保持したまま画面遷移を実現したい場合はSPAによる画面遷移が必要ですし、状態を一度リセットしたい状態で画面遷移したい場合はaタグを使うことも場合によってはあるかもしれません。

HTMLがブラウザでどのように表示されているか、SSR、CSRなどの概念をおさえておけば良いと思います :+1:

**Comment by yoshidev523** [comment]
ざっと調べましたがこれとか的を得てる記事のように感じました
https://zenn.dev/shunshimono/articles/2023-04-02_ssr-csr-mpa-spa

**Comment by yoshidev523** [comment]
> https://zenn.dev/miyaken0805/articles/c19bada3b1825a
> こんな記事ありました

上記事のaタグの説明は少し語弊があります。

> 3. Anchorタグ:
標準的な HTML の <a> タグは、新しいリクエストをサーバーに送信し、新しいページを取得する。
このプロセスはサーバーサイドで行われ、ページ全体が再読み込みされる。これにより、Next.js サーバーは新しいページを完全にレンダリングすることができる。俗に言うSSR（サーバーサイドレンダリング）ってやつ。

「標準的な HTML の <a> タグは、新しいリクエストをサーバーに送信し、新しいページを取得する。
このプロセスはサーバーサイドで行われ、ページ全体が再読み込みされる。」ここまでは正しいですが、その後の説明が正しくないです。aタグとSSRは関係ないです。SSRで画面遷移したとしてもCountのステートは保持されます（ブラウザの更新マークに変化がないです）
aタグはどちらかというとMPA(マルチページアプリケーション)での登場人物です。

私が共有した記事の
> SPA ⇄ MPA
CSR ⇄ SSR

というセットが正しいかと思います。

**Comment by yoshidev523** [comment]
この記事も参考になるかと思います（図が微妙に混乱しそうだけど。。。）

https://zenn.dev/mt877/articles/6dc3afe99ee794
